sympy/polys/matrices/ddm.py

```diff
@@ -203,6 +203,54 @@ def _check(cls, a, op, b, ashape, bshape):
             msg = "Shape mismatch: %s %s %s" % (a.shape, op, b.shape)
             raise DDMShapeError(msg)
 
+    def applyfunc(self, f, domain=None):
+        """Applies a function f to every element of a DDM
+        and converts :py:class:`~.Domain` to domain
+
+        Examples
+        ========
+
+        >>> from sympy.polys.matrices.ddm import DDM
+        >>> from sympy import ZZ, QQ
+        >>> A = DDM([[ZZ(1), ZZ(2)], [ZZ(3), ZZ(4)]], (2, 2), ZZ)
+        >>> A.applyfunc(lambda x: 2 ** x)
+        [[2, 4], [8, 16]]
+
+        We can also specify the :py:class:`~.Domain` of the output
+
+        >>> A.applyfunc(lambda x: x/QQ(5), QQ)
+        [[1/5, 2/5], [3/5, 4/5]]
```

Comment:
Rather than saying that "and converts to a domain" or "we can also specify" this should say: `If the function returns elements from a different domain then the new domain should be given as the *domain* argument.`